pkg/storer/reserve.go

```diff
@@ -184,11 +150,35 @@ func (db *DB) reserveSizeWithinRadiusWorker(ctx context.Context) {
 			db.metrics.ReserveMissingBatch.Set(float64(missing))
 		}
 
+		return count
+	}
+
+	// initial run for the metrics
+	_ = countF()
+
+	ticker := time.NewTicker(db.opts.wakeupDuration)
+	defer ticker.Stop()
+
+	for {
+
 		select {
 		case <-ctx.Done():
 			return
 		case <-ticker.C:
 		}
+
+		count := countF()
+		radius := db.StorageRadius()
+
+		if count < threshold(db.reserve.Capacity()) && db.syncer.SyncRate() == 0 && radius > 0 {
+			radius--
+			err := db.reserve.SetRadius(db.repo.IndexStore(), radius)
+			if err != nil {
+				db.logger.Error(err, "reserve set radius")
+			}
+			db.logger.Info("reserve radius decrease", "radius", radius)
```

Comment:
Would be nice to have "old_radius" and "new_radius" in the logs.

Reply:
new radius is always -> old-radius - 1